warn/warn_bazel.go

```diff
@@ -89,23 +89,29 @@ func globalVariableUsageCheck(f *build.File, category, global, alternative strin
 
 	var walk func(e *build.Expr, env *bzlenv.Environment)
 	walk = func(e *build.Expr, env *bzlenv.Environment) {
-		if ident, ok := (*e).(*build.Ident); ok {
-			if ident.Name == global {
-				if binding := env.Get(ident.Name); binding == nil {
-					if fix {
-						// It may be not correct to just replace the ident's name with `alternative` as it may be something complex
-						// like `native.package_name()` which is not a valid ident, but it's fine for reformatting.
-						ident.Name = alternative
-					} else {
-						start, end := ident.Span()
-						findings = append(findings,
-							makeFinding(f, start, end, category,
-								"Global variable \""+global+"\" is deprecated in favor of \""+alternative+"\". Please rename it.", true, nil))
-					}
-				}
-			}
+		defer bzlenv.WalkOnceWithEnvironment(*e, env, walk)
+
+		ident, ok := (*e).(*build.Ident)
+		if !ok {
+			return
+		}
+		if ident.Name != global {
+			return
+		}
+		if binding := env.Get(ident.Name); binding != nil {
+			return
+		}
+		if fix {
+			// It may be not correct to just replace the ident's name with `alternative` as it may be something complex
+			// like `native.package_name()` which is not a valid ident, but it's fine for reformatting.
+			ident.Name = alternative
```

Comment:
opt: could return here, avoid the else {}

Reply:
Done.